src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,215 @@
+ * And handle all the configurations for the creation of client.
+ */
+public final class RetryableSphereClientWithExponentialBackoff {
+    protected static final long DEFAULT_TIMEOUT = 30000;
+    protected static final long INITIAL_RETRY_DELAY = 200;
+    protected static final TimeUnit DEFAULT_TIMEOUT_TIME_UNIT = TimeUnit.MILLISECONDS;
+    protected static final int MAX_RETRIES = 5;
+    protected static final int MAX_PARALLEL_REQUESTS = 20;
+    private static final int[] DEFAULT_RETRY_ERROR_STATUS_CODES = {500, 502, 503, 504};
+
+    private SphereClientConfig clientConfig;
+    private long timeout;
+    private long initialDelay;
+    private TimeUnit timeUnit;
+    private int maxRetries;
+    private int maxParallelRequests;
+    private int[] retryErrorStatusCodes;
+
+    private RetryableSphereClientWithExponentialBackoff(@Nonnull final SphereClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+        this.timeout = DEFAULT_TIMEOUT;
+        this.timeUnit = DEFAULT_TIMEOUT_TIME_UNIT;
+        this.initialDelay = INITIAL_RETRY_DELAY;
+        this.maxRetries = MAX_RETRIES;
+        this.maxParallelRequests = MAX_PARALLEL_REQUESTS;
+        this.retryErrorStatusCodes = DEFAULT_RETRY_ERROR_STATUS_CODES;
+    }
+
+    /**
+     * Creates a new instance of {@link RetryableSphereClientWithExponentialBackoff} given a {@link SphereClientConfig}
+     * responsible for creation of a SphereClient.
+     *
+     * @param clientConfig the client configuration for the client.
+     * @return the instantiated {@link RetryableSphereClientWithExponentialBackoff}.
+     */
+    public static RetryableSphereClientWithExponentialBackoff of(
+            @Nonnull final SphereClientConfig clientConfig) {
+        return new RetryableSphereClientWithExponentialBackoff(clientConfig);
+    }
+
+    /**
+     * Sets the timeout value.
+     * @param timeout - build with timeout value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given timeout value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withTimeout(final long timeout) {
+        this.timeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the initialDelay value.
+     * @param initialDelay - build with initialDelay value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given initialDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withInitialDelay(final long initialDelay) {
+        if (initialDelay < timeout) {
+            this.initialDelay = initialDelay;
+        }
+        return this;
+    }
+
+    /**
+     * Sets the timeUnit.
+     * @param timeUnit - build with timeUnit value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given timeUnit value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withTimeUnit(@Nonnull final TimeUnit timeUnit) {
```

Comment:
Where we are using this ?

Reply:
It is not used in this class, So I have removed timeUnit from this class. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/e29bb4ec52fbac5253561b3af9312845adec7e3f